src/python/pants/core/util_rules/archive_test.py

```diff
@@ -165,3 +165,29 @@ def test_create_tar_archive(rule_runner: RuleRunner, format: ArchiveFormat) -> N
     extracted_archive = rule_runner.request(ExtractedArchive, [created_digest])
     digest_contents = rule_runner.request(DigestContents, [extracted_archive.digest])
     assert digest_contents == EXPECTED_DIGEST_CONTENTS
+
+
+@pytest.mark.parametrize(
+    "format", [ArchiveFormat.TAR, ArchiveFormat.TGZ, ArchiveFormat.TXZ, ArchiveFormat.TBZ2]
+)
+def test_create_tar_archive_raw_filename(rule_runner: RuleRunner, format: ArchiveFormat) -> None:
```

Comment:
It's not going to be obvious to future readers why this test exists. I recommend opening a bug ticket describing the regression, and then linking to it here. Also, maybe rename this function to `test_tar_in_root_dir` or something more obvious than "raw filename", as it's not clear what that refers to

Reply:
Rename 👍 

I think for the documentation it would maybe make more sense to put everything in a docstring instead of an extra issue. I can still link to this PR, which would have extra context if needed.

Reply:
Docstring is certainly a good idea!

That said, we do like to have an issue, to serve as documentation of the problem, separate from the fix. So when people search issues it's there to find. :) 

One reason linking to an issue is better than linking to a PR, is that the PR is very tied to the current state of the code. This may all get refactored in the future, so for a future reader, the PR, which will no longer reference any current code, might be confusing. An issue has a lot more longevity.

Reply:
Done

Reply:
Thanks!